pandas/core/reshape/merge.py

```diff
@@ -563,10 +563,10 @@ def __init__(
         indicator: bool = False,
         validate=None,
     ):
-        _left = _validate_operand(left)
-        _right = _validate_operand(right)
-        self.left = self.orig_left = _validate_operand(_left)  # type: "DataFrame"
-        self.right = self.orig_right = _validate_operand(_right)  # type: "DataFrame"
+        _left: "DataFrame" = _validate_operand(left)
```

Comment:
since `_validate_operand` is typed (with a return type), the variable annotation is redundant

Reply:
we dont want it just for the reader?

Reply:
i don't think so. doing this sort of thing could loosen type checking, say if for instance `_validate_operand` returned a subclass of `DataFrame` this would pass mypy and the variable would no longer have the type of the subclass

```python
import pandas as pd


class MyDataFrame(pd.DataFrame):
    pass


def some_function() -> MyDataFrame:
    return MyDataFrame()


loose: pd.DataFrame = some_function()
reveal_type(loose)
tight = some_function()
reveal_type(tight)
```

```
$ mypy test.py
test.py:13: note: Revealed type is 'pandas.core.frame.DataFrame'
test.py:15: note: Revealed type is 'test.MyDataFrame'
```

Reply:
got it, thanks.  will update